firestore/transaction_test.go

```diff
@@ -500,3 +501,42 @@ func TestRunTransaction_NonTransactionalOp(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestTransaction_WithReadOptions(t *testing.T) {
+	ctx := context.Background()
+	c, srv, cleanup := newMock(t)
+	defer cleanup()
+
+	const db = "projects/projectID/databases/(default)"
+	tm := time.Date(2021, time.February, 20, 0, 0, 0, 0, time.UTC)
+	tid := []byte{1}
+	beginReq := &pb.BeginTransactionRequest{Database: db}
+	beginRes := &pb.BeginTransactionResponse{Transaction: tid}
+
+	srv.reset()
+	srv.addRPC(beginReq, beginRes)
+
+	srv.addRPC(
+		&pb.CommitRequest{
+			Database:    db,
+			Transaction: tid,
+		},
+		&pb.CommitResponse{CommitTime: aTimestamp3},
+	)
+
+	srv.addRPC(
+		&pb.CommitRequest{
+			Database:    db,
+			Transaction: tid,
+		},
+		&pb.CommitResponse{CommitTime: aTimestamp3},
+	)
+
+	if err := c.RunTransaction(ctx, func(ctx2 context.Context, tx *Transaction) error {
+		docref := c.Collection("C").Doc("a")
+		tx.WithReadOptions(ReadTime(tm)).Get(docref)
```

Comment:
What's the relationship between tm and aTimestamp3 here?

Reply:
Clarified.